mucore/src/muBase.c

```diff
@@ -1226,3 +1226,81 @@ MU_VOID muDebugError(muError_t errorcode)
 	}
 
 }
+
+/****************************************************************************************\
+ *          Draw Rectangle function                                                      *
+ \****************************************************************************************/
+/* Draw Rectangle */
+
+muError_t muRectangle(muImage_t *SrcImg, muPoint_t p1, muPoint_t p2, MU_8S color)
+{
+	MU_32S i, j;
+	MU_32S offset, offset2;
+	MU_32S Ch_tmp;
+	MU_32S Ch_width;
+	MU_8U b = 0, g = 0, r = 0;
+
+	Ch_width = SrcImg->width*SrcImg->channels;
+
+	Ch_tmp = p2.x * SrcImg->channels;
+

```

Comment:
if this is only for BMP format. please check the channel like below:
Even the index in while loop is under control.
	if(SrcImg->channels != 3)
	{
		return MU_ERR_NOT_SUPPORT;
	}